hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/BlockStreamTransactionTranslator.java

```diff
@@ -0,0 +1,290 @@
+        return recordBuilder;
+    }
+
+    private TransactionRecord.Builder parseTransactionResult(
+            final BlockItem txnBlockItem,
+            final TransactionRecord.Builder recordBuilder,
+            final TransactionReceipt.Builder receiptBuilder) {
+        final var txnResult = txnBlockItem.transactionResult();
+
+        recordBuilder.automaticTokenAssociations(txnResult.automaticTokenAssociations());
+        recordBuilder.parentConsensusTimestamp(txnResult.parentConsensusTimestamp());
+        recordBuilder.consensusTimestamp(txnResult.consensusTimestamp());
+        recordBuilder.scheduleRef(txnResult.scheduleRef());
+        recordBuilder.paidStakingRewards(txnResult.paidStakingRewards());
+        recordBuilder.transactionFee(txnResult.transactionFeeCharged());
+        recordBuilder.transferList(txnResult.transferList());
+        recordBuilder.tokenTransferLists(txnResult.tokenTransferLists());
+
+        receiptBuilder.exchangeRate(txnResult.exchangeRate());
+        receiptBuilder.status(txnResult.status());
+
+        return recordBuilder;
+    }
+
+    private TransactionRecord.Builder parseTransactionOutput(
+            final BlockItem transactionBlockItem, final TransactionRecord.Builder trb, TransactionReceipt.Builder rb) {
+        final var txnOutput = transactionBlockItem.transactionOutput();
+
+        // TODO: why are so many of these methods missing?
+        //            if (txnOutput.hasCryptoCreate()) {
+        //                rb.accountID(txnOutput.cryptoCreate().accountID());
+        //                trb.alias(txnOutput.cryptoCreate().alias());
+        //            }
+
+        if (txnOutput.hasCryptoTransfer()) {
+            trb.assessedCustomFees(txnOutput.cryptoTransfer().assessedCustomFees());
+        }
+        //            if (txnOutput.hasFileCreate()) {
+        //                rb.fileID(txnOutput.fileCreate().fileID());
+        //            }
+
+        if (txnOutput.hasContractCreate()) {
+            rb.contractID(txnOutput.contractCreate().contractCreateResult().contractID());
+        }
+
+        if (txnOutput.hasContractCreate()) {
+            trb.contractCreateResult(txnOutput.contractCreate().contractCreateResult());
+        }
+        if (txnOutput.hasContractCall()) {
+            trb.contractCallResult(txnOutput.contractCall().contractCallResult());
+        }
+
+        if (txnOutput.hasEthereumCall()) {
+            trb.ethereumHash(txnOutput.ethereumCall().ethereumHash());
+        }
+
+        //            if (txnOutput.hasTopicCreate()) {
+        //                rb.topicID(txnOutput.topicCreate().topicID());
+        //            }
+
+        if (txnOutput.hasSubmitMessage()) {
+            //                rb.topicSequenceNumber(txnOutput.submitMessage().topicSequenceNumber());
+            rb.topicRunningHashVersion(
+                    txnOutput.submitMessage().topicRunningHashVersion().protoOrdinal());
```

Comment:
It looks like the `protoOrdinal()` has values in reverse order i.e. 0 is the most recent version, while services handler uses `RUNNING_HASH_VERSION = 3L`, is that correct?

Reply:
From the comments in the `RunningHashVersion` protobuf, the `protoOrdinal()`, it's correct that 0 is the most recent version, and it does appear that the submit message handler uses 3L.

This code is only taking the output and setting it on the record builder, so it should be correct in either case.

Reply:
Yeah, actually there's a [TODO for block stream writing](https://github.com/hashgraph/hedera-services/blob/stream-manager-spike/hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/IoBlockItemsBuilder.java#L702) for to be confirmed for the value of the field, so if there is going to be some change it should be addressed there.